nextmv/nextmv/local/executor.py

```diff
@@ -0,0 +1,565 @@
+        outputs_dir=outputs_dir,
+        stdout_output=stdout_output,
+    )
+    process_run_solutions(
+        run_id=run_id,
+        run_dir=run_dir,
+        temp_run_outputs_dir=temp_run_outputs_dir,
+        temp_src=temp_src,
+        outputs_dir=outputs_dir,
+        stdout_output=stdout_output,
+    )
+    process_run_visuals(
+        run_dir=run_dir,
+        outputs_dir=outputs_dir,
+    )
+
+
+def process_run_information(run_id: str, run_dir: str, result: subprocess.CompletedProcess[str]) -> None:
+    """
+    Processes the run information, updating properties such as duration and
+    status.
+
+    Parameters
+    ----------
+    run_id : str
+        The ID of the run.
+    run_dir : str
+        The path to the run directory.
+    result : subprocess.CompletedProcess[str]
+        The result of the subprocess run.
+    """
+
+    info_file = os.path.join(run_dir, f"{run_id}.json")
+
+    with open(info_file) as f:
+        info = json.load(f)
+
+    # Calculate duration.
+    created_at_str = info["metadata"]["created_at"]
+    created_at = datetime.fromisoformat(created_at_str.replace("Z", "+00:00"))
+    now = datetime.now(timezone.utc)
+    duration = round((now - created_at).total_seconds() * 1000, 1)
+
+    # Update the status
+    status = StatusV2.succeeded.value
+    error = ""
+    if result.returncode != 0:
+        status = StatusV2.failed.value
+        error = result.stderr if result.stderr else "unknown error"
+
+    # Update the run info file.
+    info["metadata"]["duration"] = duration
+    info["metadata"]["status_v2"] = status
+    info["metadata"]["error"] = error
+
+    with open(info_file, "w") as f:
+        json.dump(info, f, indent=2)
+
+
+def process_run_logs(run_dir: str, result: subprocess.CompletedProcess[str]) -> None:
+    """
+    Processes the logs of the run. Writes the logs to a logs directory.
+
+    Parameters
+    ----------
+    run_dir : str
+        The path to the run directory.
+    result : subprocess.CompletedProcess[str]
+        The result of the subprocess run.
+    """
+
+    logs_dir = os.path.join(run_dir, LOGS_KEY)
+    os.makedirs(logs_dir, exist_ok=True)
+    with open(os.path.join(logs_dir, LOGS_FILE), "w") as f:
+        f.write(result.stderr)
+
+
+def process_run_statistics(temp_run_outputs_dir: str, outputs_dir: str, stdout_output: dict[str, Any]) -> None:
+    """
+    Processes the statistics of the run. Check for an outputs/statistics folder
+    being created by the run. If it exists, copy it to the run directory. If it
+    doesn't exist, attempt to get the stats from stdout.
+
+    Parameters
+    ----------
+    temp_run_outputs_dir : str
+        The path to the temporary outputs directory.
+    outputs_dir : str
+        The path to the outputs directory in the run directory.
+    stdout_output : dict[str, Any]
+        The stdout output of the run, parsed as a dictionary.
+    """
+
+    stats_src = os.path.join(temp_run_outputs_dir, STATISTICS_KEY)
```

Comment:
I think in case of `multi-file` statistics are just one file (which location can be configured now - live in prod). So, I don't think a directory works here, right?
For `csv-archive` and `json` just checking the `statistics` key makes sense. I am not sure whether ignoring the content type of the run is stable enough, but might be fine.

Reply:
Adapted for multi-file to use the configurations in the manifest if they are defined or defaulting to this when not.